net/DescriptionViewController.m

```diff
@@ -86,104 +174,527 @@ -(void)issueImageDidLoad
+        //error
+    }];
+}
+
+-(void)sendMessage:(NSString*)message andAddnewCommentsBlockWithAllUserDictionaries:(NSArray<NSDictionary<NSString *,NSString *> *> *) allUserDictionaries
+{
+    __weak DescriptionViewController *weakSelf = self;
+    
+    [self.dataSorce requestSendNewComment:message
+    forIssueID:[CurrentItems sharedItems].issue.issueId
+    andHandler:^(NSArray<NSDictionary<NSString *,id> *> *commentDics, NSError *error) {
+        dispatch_async(dispatch_get_main_queue(), ^{
+           if(commentDics==nil || error != nil)
+               return;
+           
+           for (NSInteger index=self.commentBoxArr.count; index<commentDics.count; ++index)
+           {
+               
+               NSDictionary<NSString *,id> *commentDic = commentDics[index];
+               [weakSelf addOneComment:commentDic withIndex:index allUsersDic:allUserDictionaries];
+           }
+           NSLog(@"weakSelf.viewBetweenCommentAndChare %@",weakSelf.viewBetweenCommentAndChare);
+           weakSelf.contentViewHeightConstraint.constant = weakSelf.contentDynamicHeight + weakSelf.contentStaticHeight;
+           [weakSelf.view layoutIfNeeded];
+            
+            
+            CGFloat yOffset = self.contentView.frame.size.height - self.ScrollView.frame.size.height;
+            if (yOffset>0)
             {
-                [self makeCategoryLabelWithStringCategory:issueCategory.name];
-                break;
+                [UIView animateWithDuration:0.3 animations:^{
+                    self.ScrollView.contentOffset = CGPointMake(0, yOffset);
+                }];
             }
-        }
-        
+        });
+    }];
+    
+}
+
+
+
+#pragma mark - Dynamic elements
+
+-(void)requestUsersAndComments
+{
+    __weak DescriptionViewController *weakSelf = self;
+    [self.dataSorce requestAllUsers:^(NSArray<NSDictionary<NSString *,NSString *> *> *userDictionaries) {
+        [weakSelf commentBlockWithAllUserDictionaries:userDictionaries];
     } withErrorHandler:^(NSError *error) {
-        // TODO: handle error
+        //error
+    }];
+}
+
+
+
+-(void)commentBlockWithAllUserDictionaries:(NSArray<NSDictionary<NSString *,NSString *> *> *) allUserDictionaries
+{
+    __weak DescriptionViewController *weakSelf = self;
+    self.viewToConnectDynamicItems = self.viewBetweenCommentAndChare;
+    
+    [self.dataSorce requestCommentsWithIssueID:[CurrentItems sharedItems].issue.issueId
+                                    andHandler:^(NSArray<NSDictionary<NSString *,id> *> *commentDics, NSError *error) {
+       dispatch_async(dispatch_get_main_queue(), ^{
+           weakSelf.contentViewHeightConstraint.constant =  weakSelf.contentStaticHeight;
+           [weakSelf.view layoutIfNeeded];
+           weakSelf.contentDynamicHeight = 0;
+           if(commentDics==nil || error != nil || [commentDics isKindOfClass: [NSDictionary class]])
+               return;
+           weakSelf.contentView.restorationIdentifier = @"contentView";
```

Comment:
It does'n have any sence more. I've created it for debugging constraints. I'll remove it.